src/Tickets/QR/Controller.php

```diff
@@ -27,21 +27,108 @@ public function is_active(): bool {
 	/**
 	 * Register the controller.
 	 *
-	 * @since   TBD
+	 * @since TBD
+	 *
+	 * @uses  Notices::register_admin_notices()
+	 *
+	 * @return void
 	 */
 	public function do_register(): void {
 		$this->container->bind( QR::class, [ $this, 'bind_facade_or_error' ] );
+		$this->container->singleton( Settings::class, Settings::class );
+		$this->container->singleton( Notices::class, Notices::class );
+		$this->container->singleton( Connector::class, Connector::class );
+		$this->container->singleton( Observer::class, Observer::class );
```

Comment:
Since the implementation is same as class name, the 2nd arg isn't necessary.
```suggestion
		$this->container->singleton( Settings::class );
		$this->container->singleton( Notices::class );
		$this->container->singleton( Connector::class );
		$this->container->singleton( Observer::class );
```